src/Control/Monad/Free/Compat.purs

```diff
@@ -0,0 +1,82 @@
+module Control.Monad.Free.Compat
```

Comment:
An alternative to this module would be to keep these functions in the original module, and add deprecation notices to them ("in the next major version this function will be renamed to <x>"). Do y'all have a preference between those two approaches?